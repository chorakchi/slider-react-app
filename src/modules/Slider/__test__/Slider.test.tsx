
import React from "react";
import { shallow, mount } from "enzyme";
import * as renderer from "react-test-renderer";
import {Slider} from "./../index";

describe("Slider", () => {
  const defaultProps = {
    data: ['Image-1','Image-2'],
  };

  it("should render the Slider", () => {
    const subject = shallow(<Slider {...defaultProps} />);
    expect(subject).toHaveLength(1);
  });

  it("should render correctly and match to the Snapshot", () => {
    const subject = renderer.create(<Slider {...defaultProps} />).toJSON();
    expect(subject).toMatchSnapshot();
  });

    it("should call onChangeSelector and rednder correctly and have a key on selecting Burger", () => {
    const mockCallback = jest.fn(() => "i return something!");
    const subject = mount(
      <Slider {...defaultProps} />
    );
    const backButton = subject.find("IconButton").filterWhere(item => {
      return item.prop("id") === "btn-back";
    });
    backButton.simulate("click");
    backButton.simulate("click");
        const nextButton = subject.find("IconButton").filterWhere(item => {
      return item.prop("id") === "btn-next";
    });
    nextButton.simulate("click");
    nextButton.simulate("click");
        const indicatorButton = subject.find("#Indicator-1").at(1);
    indicatorButton.simulate("click");
  });
});