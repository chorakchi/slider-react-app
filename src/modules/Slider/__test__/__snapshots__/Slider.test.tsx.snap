// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider should render correctly and match to the Snapshot 1`] = `
<div
  className="sc-jSFkmK hGelrT"
>
  <div
    className="sc-bdnylx bMWfWs"
    id="btn-back"
    onClick={[Function]}
  >
    <svg
      style={
        Object {
          "height": 24,
          "width": 24,
        }
      }
      viewBox="0 0 24 24"
    >
      <path
        d="M15.41,16.58L10.83,12L15.41,7.41L14,6L8,12L14,18L15.41,16.58Z"
        fill="currentColor"
      />
    </svg>
  </div>
  <div>
    <div
      className="sc-hKFyIo kIFwax"
    >
      <img
        className="sc-eCApGN bVoWZr"
        src="Image-1"
      />
      <img
        className="sc-eCApGN bVpKby"
        src="Image-2"
      />
    </div>
    <div
      className="sc-gtssRu dMrldS"
    >
      <div
        className="sc-dlnjPT FdIJO"
        id="Indicator-0"
        onClick={[Function]}
        selected={true}
      />
      <div
        className="sc-dlnjPT bhIUTS"
        id="Indicator-1"
        onClick={[Function]}
        selected={false}
      />
    </div>
  </div>
  <div
    className="sc-bdnylx bMWfWs"
    id="btn-next"
    onClick={[Function]}
  >
    <svg
      style={
        Object {
          "height": 24,
          "width": 24,
        }
      }
      viewBox="0 0 24 24"
    >
      <path
        d="M8.59,16.58L13.17,12L8.59,7.41L10,6L16,12L10,18L8.59,16.58Z"
        fill="currentColor"
      />
    </svg>
  </div>
</div>
`;
